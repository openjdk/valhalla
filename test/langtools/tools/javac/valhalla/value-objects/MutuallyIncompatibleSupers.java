/*
 * @test /nodynamiccopyright/
 * @bug 8287767
 * @summary [lw4] Javac tolerates mutually incompatible super types.
 * @compile/fail/ref=MutuallyIncompatibleSupers.out -XDrawDiagnostics -XDdev MutuallyIncompatibleSupers.java
 */

public class MutuallyIncompatibleSupers {

    identity interface II {}
    value interface VI {}

    static abstract class X implements II, VI {} // mutually incompatible supers.

    interface GII extends II {} // OK.
    value interface BVI extends GII {} // Error

    interface GVI extends VI {} // OK.
    identity interface BII extends GVI {} // Error

    static value class BVC implements II {} // Error
    class BIC implements VI {} // Error
}
