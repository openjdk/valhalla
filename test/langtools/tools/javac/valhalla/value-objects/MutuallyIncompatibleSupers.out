MutuallyIncompatibleSupers.java:13:21: compiler.err.mutually.incompatible.supers: MutuallyIncompatibleSupers.X, MutuallyIncompatibleSupers.II, MutuallyIncompatibleSupers.VI
MutuallyIncompatibleSupers.java:16:11: compiler.err.value.type.has.identity.super.type: MutuallyIncompatibleSupers.BVI, MutuallyIncompatibleSupers.II
MutuallyIncompatibleSupers.java:19:14: compiler.err.identity.type.has.value.super.type: MutuallyIncompatibleSupers.BII, MutuallyIncompatibleSupers.VI
MutuallyIncompatibleSupers.java:21:18: compiler.err.value.type.has.identity.super.type: MutuallyIncompatibleSupers.BVC, MutuallyIncompatibleSupers.II
MutuallyIncompatibleSupers.java:22:5: compiler.err.identity.type.has.value.super.type: MutuallyIncompatibleSupers.BIC, MutuallyIncompatibleSupers.VI
5 errors
